### PR TITLE
added splash screen for android- #2756

### PR DESCRIPTION
**Pull request by prateek3255**
This PR is targeted to solve the issue #2756 

I updated a few files and added a new package `react-native-splash-screen` for adding splash screen for the android part( I couldn't do it for the ios part because I don't have a mac ) following [this](https://medium.com/handlebar-labs/how-to-add-a-splash-screen-to-a-react-native-app-ios-and-android-30a3cec835ae) blog.

Here is the gif on my device for the same -

zulip_splash

### CODE REVIEW

**Comment by borisyankov** [comment]
I am wondering if we need to use `react-native-splash-screen`?
Android and iOS both have capabilities to add the splash screen.
I am always cautious when adding new dependencies because when they break or become incompatible with the latest React Native releases.

File: src/ZulipMobile.js [hunk 1]
```diff
@@ -9,25 +9,33 @@ import CompatibilityChecker from './boot/CompatibilityChecker';
...
-            <StylesProvider>
-              <AppWithNavigation />
-            </StylesProvider>
-          </TranslationProvider>
-        </AppDataFetcher>
-      </AppEventHandlers>
-    </StoreProvider>
-  </CompatibilityChecker>
-);
+export default class ZulipMobile extends PureComponent<Props> {
```
Comment (jainkuniya) [inline src/ZulipMobile.js]: I guess changing this to 
```
export default class ZulipMobile extends PureComponent<void> 
```
will fix flow issue.

**Comment by prateek3255** [comment]
@borisyankov this is probably the most easiest and best possible way to do it and this dependency is only required for the android part only because we need 2 splash screens there, 1 in native android code and the other one using the `react-native-splash-screen`, if we remove the other one there will be delay with a white screen between the android splash screen and the actual loaded app, you can find more details in [this](https://medium.com/handlebar-labs/how-to-add-a-splash-screen-to-a-react-native-app-ios-and-android-30a3cec835ae) blog. 

And `react-native-splash-screen` has around 2204 stars and the last commit was also only 3 days ago, so I don't think there will be any compatibility issue.

**Comment by prateek3255** [comment]
Sorry for making a cluster of commits here, I am a newbie to open source and the CI tests, should I squash them into one?

**Comment by gnprice** [comment]
Thanks @prateek3255 , and welcome! I haven't yet gotten a chance to review this code, but to your question about Git: our practice is to make each commit [clear and coherent](https://zulip.readthedocs.io/en/latest/contributing/version-control.html). So commits that just fix up test breakage in previous commits (like here, it looks like) should definitely be squashed in. On the other hand, when there are several logical changes that can be done separately, they're best as a series of separate commits.

Also, you can run the tests yourself before submitting -- no need to wait for Travis CI:
```
yarn test
```

**Comment by prateek3255** [comment]
Can anyone please review my PR?

**Comment by gnprice** [comment]
Thanks @prateek3255 for sending this!

@borisyankov and I just took a quick look at that library -- and it's a pretty small amount of code. Here's the file that looks like the actual splash screen, on Android:
https://github.com/crazycodeboy/react-native-splash-screen/blob/master/android/src/main/java/org/devio/rn/splashscreen/SplashScreen.java

When the library is this small, our experience has been
* Like most things, it's probably not quite right.
* To fix the ways that it's not quite right, we're going to have to end up reading the respective Android and iOS documentation and figuring out how things are actually supposed to work.
* Once we do, it'll have been easier all around if we just write the code ourselves based on reading the docs :slightly_smiling_face: Because the library is so small, it's likely that the author themself didn't do much more than copy the relevant sample code from the platform docs + the RN docs, and put them together -- so we can do that instead.

I'm going to close this PR, but we'd definitely like to have a splash screen for the reasons you say. If you'd like to take that on for Android -- reading Android and Android Studio docs, explaining what you find, and using Android Studio to make a splash screen -- I'd be glad to see that PR. :smile:

**Pull request closed without merging**